src/manage_sql.c

```diff
@@ -44510,68 +44494,66 @@ modify_filter (const char *filter_id, const char *name, const char *comment,
 
   if ((filter_in_use_for_output (filter)
        || filter_in_use_for_result_event (filter))
-      && f_type
-      && strcasecmp (f_type, "result"))
+      && type
+      && strcasecmp (type, "result"))
     {
-      g_free (f_name);
-      g_free (f_comment);
-      g_free (f_type);
-      g_free (f_term);
       sql_rollback ();
       return 5;
     }
 
   if (filter_in_use_for_secinfo_event (filter)
-      && f_type
-      && strcasecmp (f_type, "info"))
+      && type
+      && strcasecmp (type, "info"))
     {
-      g_free (f_name);
-      g_free (f_comment);
-      g_free (f_type);
-      g_free (f_term);
       sql_rollback ();
       return 6;
     }
 
   /* Check whether a filter with the same name exists already. */
-  if (f_name)
+  if (name)
     {
-      if (resource_with_name_exists (f_name, "filter", filter))
+      if (resource_with_name_exists (name, "filter", filter))
         {
-          g_free (f_name);
-          g_free (f_comment);
-          g_free (f_type);
-          g_free (f_term);
           sql_rollback ();
           return 2;
         }
     }
 
-  quoted_name = sql_quote(f_name ?: "");
-  clean_term = manage_clean_filter (f_term ? f_term : "",
+  quoted_filter_id = sql_quote(filter_id);
+  quoted_name = sql_quote(name ?: "");
+  clean_term = manage_clean_filter (term ? term : "",
                                     0 /* ignore_max_rows_per_page */);
   quoted_term = sql_quote (clean_term);
   g_free (clean_term);
-  quoted_comment = sql_quote (f_comment ? f_comment : "");
-  quoted_type = sql_quote (f_type ? f_type : "");
+  quoted_comment = sql_quote (comment ? comment : "");
+  quoted_type = sql_quote (type ? type : "");
+
+  t_name = name ? g_strdup_printf (", name = '%s'", quoted_name) :
+                  g_strdup ("");
+  t_comment = comment ? g_strdup_printf (", comment = '%s'", quoted_comment) :
+                        g_strdup ("");
+  t_term = term ? g_strdup_printf (", term = '%s'", quoted_term) :
+                  g_strdup ("");
+  t_type = type ? g_strdup_printf (", type = lower ('%s')", quoted_type) :
+                  g_strdup ("");
 
   sql ("UPDATE filters SET"
-       " name = '%s',"
-       " comment = '%s',"
-       " term = '%s',"
-       " type = lower ('%s'),"
-       " modification_time = m_now ()"
+       " uuid = '%s'"
```

Comment:
The `uuid` column already has this value, because we get `filter` from `filter_id` above (using `find_filter_with_permission`). The `uuid` column must always stay the same anyway.